PLAN-2443 - Include allReduce in the Pattern docs where QPPs are described (#1407)
neo4j/docs-cypher: modules/ROOT/pages/patterns/variable-length-patterns.adoc

* Nodes or relationships must have properties satisfying some condition.
For example, all relationships must have the property `distance > 10`.

> * For every iteration of the QPP, an aggregated value over the constructed path so far should satisfy a predicate.

Reviewer: ```suggestion
* For every iteration of the quantified path pattern, an aggregated value over the constructed path so far should satisfy a predicate.
```

> * For every iteration of the quantified path pattern, an aggregated value over the constructed path so far should satisfy a predicate.

Reviewer 2: ```suggestion
* For every iteration of the quantified path pattern, an aggregated value over the constructed path so far must satisfy a predicate.
```

double-check if this reflects the intended meaning

It also shows that there is only one path to solving the query (a number that remains constant even if the data from the rest of the UK railway network was included).
Using inline predicates or making quantified path patterns more specific where possible can thus greatly improve query performance.

An alternative is to use an upper bound for the total distance, for example `6.05`.

> Once a path exceeds this upper bound, we can prune it and continue searching other paths.

Reviewer 2: ```suggestion
Once a path exceeds this upper bound, you can prune it and continue searching other paths.
```

It also shows that there is only one path to solving the query (a number that remains constant even if the data from the rest of the UK railway network was included).
Using inline predicates or making quantified path patterns more specific where possible can thus greatly improve query performance.

An alternative is to use the information that we want to find paths with a total distance less than `6.05`.

> From the `ALL SHORTEST` query, we know that there exists a path with a total distance of `6.04`.

Reviewer: I would avoid mentioning SHORTEST here, it muddies the water. We don't want to suggest the association of `SHORTEST` with `allReduce()`.
Like in the previous example we say "This works fine where the solution is known to lie within some range of hops.", here I would say that I pick a known upper bound of `6.05` or maybe even `7` or `10` for the total distance. I wouldn't specify how I know it to be a reasonable upper bound.
